docs/manual/configuration-options.rst

```diff
@@ -122,6 +122,32 @@ application                         **REQUIRED**                        The name
 secret                              **REQUIRED**                        The secret of the approved application (basic authentication password).
 =================================== ==================================  =============================================================================
 
+.. _configuration-hash:
+
+User Password Hashing
+=====================
+
+This configuration object is **OPTIONAL**.
+
+This group of options allows you to configure the hashing algorithm used by Thunder for server-side hashing of
+user passwords, as well as the algorithm used to check the password value in the request header.
+
+.. code-block:: yaml
+
+    passwordHash:
+      algorithm:
+      serverSideHash:
+
+
+=================================== ==================================  =============================================================================
+Name                                Default                             Description
+=================================== ==================================  =============================================================================
+algorithm                           simple                              The algorithm to use for server side hashing and password comparison.
+                                                                        Supported values are: ``simple``, ``md5``, and ``bcrypt``.
```

Comment:
Are we supporting sha2? Also if server side hashing is for storing results in the DB then maybe we should only support algorithms that use a salt.

Reply:
I will add support for sha2 in a future PR. We can add support for as many as we want, I've made it pretty easy to add more!

I agree in principle what you're saying about only supporting algorithms with a salt. Definitely for our own use we should be using bcrypt or something similar. I do want to keep md5 support in here just in case someone wants it. It also allows for easier unit testing to have an option without a salt since you know what the value is going to be when hashed. You'll see I use the MD5 algorithm in the unit tests for `UserResource`.

In the end, I think more user choice is ok. They can even choose the `simple` algorithm which doesn't do any hashing (lol)